### ISSUE TITLE
Terraform crash (SIGSEGV) when creating an aws_elasticsearch_domain against Localstack

### ISSUE THREAD

**Issue by davidemurray-nike** [terraform-providers/terraform-provider-aws#7638]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11

### Affected Resource(s)

* aws_elasticsearch_domain

### Terraform Configuration Files

```hcl
resource "aws_elasticsearch_domain" "asset_metadata_localstack" {
  domain_name           = "my-domain"
  elasticsearch_version = "6.3"
}
```

### Panic Output

https://gist.github.com/davidemurray-nike/1ad4db25a92d5a285a628e5c50ad7dc7

### Expected Behavior

An ElasticSearch domain should have been created without error.

### Actual Behavior

An ElasticSearch domain is created, but Terraform crashes and the domain is not in its state, so there's an error about an existing domain the next time Terraform apply is run.

### Steps to Reproduce

1. Start [Localstack](https://github.com/localstack/localstack).
2. Create a provider suited for Localstack like the following:

```
provider "aws" {
  region                      = "${var.region}"
  skip_credentials_validation = true
  skip_requesting_account_id  = true
  skip_metadata_api_check     = true
  s3_force_path_style         = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
  version                     = "<=1.29.0"

  endpoints {
    dynamodb = "http://localhost:4569"
    s3       = "http://localhost:4572"
    es       = "http://localhost:4578"
  }
}
```

3. `terraform apply`

### Important Factoids

My Terraform does not specify coginito_options, but I get the same SIGSEGV error if I specify cognito_options, both with enabled = false and dummy values for the required fields and with enabled = true and dummy values.

The only resolution I've found is to use version `1.29.0` of the aws provider, the version before function `github.com/terraform-providers/terraform-provider-aws/aws.flattenESCognitoOptions` was introduced.

### References

* #0000

**Comment by bflad** [comment]
Hi @davidemurray-nike 👋 Thank you for the thorough issue report and sorry for the unexpected behavior here. I was able to reproduce this as well.

We don't currently run our acceptance testing suite against Localstack, but we do try to provide best effort support for the local development environment to at least prevent Terraform crashes. Luckily in this case it looks like it'll be fairly easy to fix up by ensuring `flattenESCognitoOptions` performs a `nil` check on the incoming `CognitoOptions` before attempting to access it.

The maintainers are heads down for version 2.0.0 development and testing work for the next week or so, but hopefully we can get to this shortly thereafter (unless a community member beats us on the implementation 😄).

**Comment by paultyng** [comment]
This is a common issue I've encountered in local stack, its implementation of API endpoints does not currently fulfill all aspects of the AWS JSON so we get a lot of `nil`'s where they are unexpected. We can add additional `nil` checks in the provider and work on getting this able to run acc tests against localstack, but its a pretty significant undertaking. It would probably be good to also fix this upstream in localstack. 

I've opened 7921 to look further in to running acceptance tests (and supporting TF test runs) against localstack.
